src/rgw/driver/rados/rgw_rados.cc

```diff
@@ -5322,6 +5326,7 @@ int RGWRados::restore_obj_from_cloud(RGWLCCloudTierCtx& tier_ctx,
     }
     //set mtime to now()
     set_mtime = real_clock::now();
```

Comment:
For both transition and restore (temp & permanent)  mtime should remain constant and only internal_mtime should be updated.